### PR TITLE
factorize readPath

### PR DESCRIPTION
**Pull request by zvin**

### CODE REVIEW

**Comment by lurch** [comment]
Removing myself, as I'm not very good at reading Promise-based code.

File: build/boot-record.js [hunk 1]
```diff
@@ -39,13 +39,9 @@ exports.read = function(image, position) {
...
 readPath = function(image, position) {
-  var result;
-  result = new Buffer(BOOT_RECORD_SIZE);
-  return fs.openAsync(image, 'r').then(function(fd) {
-    return fs.readAsync(fd, result, 0, BOOT_RECORD_SIZE, position)["return"](fd);
-  }).then(function(fd) {
-    return fs.closeAsync(fd);
-  })["return"](result);
+  return Promise.using(filedisk.openFile(image, 'r'), function(fd) {
+    return readFileDisk(new filedisk.FileDisk(fd), position);
```
Comment (jhermsmeier) [inline build/boot-record.js:43]: Is the intention to leave the `fd` unclosed, or am I missing something?

Comment (zvin) [reply build/boot-record.js:43]: @jhermsmeier `filedisk.openFile` is a disposer.
Used with `Promise.using` it guarantees that the fd will be closed.
See https://github.com/resin-io-modules/file-disk/blob/master/index.js#L47-L58 and http://bluebirdjs.com/docs/api/promise.using.html

**Review by jviotti — APPROVED** [review]

File: build/boot-record.js (hunk 1 above)
Comment (jhermsmeier) [reply build/boot-record.js:43]: @zvin I see, thanks for the explanation!

**Pull request merged**